File: 백준/Bronze/2798. 블랙잭/블랙잭.py
import sys

n, m = map(int, sys.stdin.readline().split())
arr = list(map(int, sys.stdin.readline().split()))
max_num = 0

for i in range(n-2):
    for j in range(i+1, n-1):
        for k in range(j+1, n):
            tmp = arr[i] + arr[j] + arr[k]

            if tmp <= m and tmp > max_num:
                max_num = tmp

sys.stdout.write(f"{max_num}\n")

File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    int n, m;
    int max_num = 0;
    cin >> n >> m;
    vector<int> arr(n);

    for (int& num : arr)
        cin >> num;

    for (int i = 0; i < n-2; i++) {
        for (int j = i+1; j < n-1; j++) {
            for (int k = j+1; k < n; k++) {
                int tmp = arr[i] + arr[j] + arr[k];
                
                if (tmp <= m && tmp > max_num)
                    max_num = tmp;
            }
        }
    }

    cout << max_num << "\n";
        
    return 0;
}